pkg/checkversion.go

```diff
@@ -0,0 +1,84 @@
+package pkg
+
+import (
+	"github.com/Masterminds/semver"
+	"github.com/ghodss/yaml"
+	log "github.com/sirupsen/logrus"
+	"io/ioutil"
+	"net/http"
+	"sort"
+	"strings"
+)
+
+const chartRepositoryUrl = "https://cronitorio.github.io/cronitor-kubernetes/index.yaml"
+
+type chartEntry struct {
+	ApiVersion  string   `yaml:"ApiVersion"`
+	AppVersion  string   `yaml:"AppVersion"`
+	Created     string   `yaml:"created"`
+	Description string   `yaml:"description"`
+	Digest      string   `yaml:"digest"`
+	Name        string   `yaml:"name"`
+	Urls        []string `yaml:"urls"`
+	Version     string   `yaml:"version"`
+}
+
+type chart struct {
+	ApiVersion string                  `yaml:"ApiVersion"`
+	Generated  string                  `yaml:"generated"`
+	Entries    map[string][]chartEntry `yaml:"Entries"`
+}
+
+func getChartYaml() ([]byte, error) {
+	resp, err := http.Get(chartRepositoryUrl)
+	if err != nil {
+		return nil, err
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	defer resp.Body.Close()
+	if err != nil {
+		log.Fatalln(err)
+	}
+	return body, nil
+}
+
+func extractVersionsFromChart(chartBody []byte) ([]string, error) {
+	var data chart
+	if err := yaml.Unmarshal(chartBody, &data); err != nil {
+		return nil, err
+	}
+
+	var versions []string
+	for _, row := range data.Entries["cronitor-kubernetes"] {
+		versions = append(versions, row.AppVersion)
+	}
+	return versions, nil
+}
+
+func extractLatestVersionFromList(versions []string) string {
+	vs := make([]*semver.Version, len(versions))
+	for i, r := range versions {
+		v, err := semver.NewVersion(r)
+		if err != nil {
+			log.Errorf("Error parsing version: %s", err)
+		}
+
+		vs[i] = v
+	}
+	sort.Sort(semver.Collection(vs))
+	lastItem := vs[len(vs)-1]
+	return lastItem.String()
+}
+
+func GetLatestVersion() string {
+	data, err := getChartYaml()
+	if err != nil {
+		log.Errorf("%v", err)
```

Comment:
If there's an error fetching the chart from the repo, wouldn't this swallow it and then it will fail when trying to use the data?

Maybe my Go is a bit rusty :)

Reply:
Yes, but that's okay. This isn't a critical thing that has to happen so we'd rather swallow and move on.

Reply:
Ah ok, then all good! I thought it would panic when you try to access the data or currentVersion in case of an error.